src/hooks/useAuthCheck.ts

```diff
@@ -0,0 +1,209 @@
+      if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
+        const skipChecksWhenExpired = true;
+        if (skipChecksWhenExpired) {
+          // Skip frequent checks when already in expired state
+          return;
+        }
+      }
+
+      lastCheckRef.current = now;
+      const status = await getAuthStatus();
+
+      if (status === AUTH_STATUS.HASH_KEY_EXPIRED) {
+        navigateToLockScreen();
+      }
+    } catch (error) {
+      logger.error(
+        "useAuthCheck.checkAuth",
+        "Failed to check auth status",
+        error,
+      );
+    }
+  }, [getAuthStatus, navigateToLockScreen, authStatus]);
+
+  // Set up interval for checking auth based on app state
+  const setupCheckInterval = useCallback(
+    (state: AppStateStatus) => {
+      // Clear any existing interval
+      if (checkIntervalRef.current) {
+        clearInterval(checkIntervalRef.current);
+        checkIntervalRef.current = null;
+      }
+
+      let intervalTime: number;
+
+      // Determine check frequency based on app state
+      if (state === "active") {
+        if (isActive) {
+          // User actively using the app
+          intervalTime = ACTIVE_CHECK_INTERVAL;
+        } else {
+          // App in foreground but user not actively interacting
+          intervalTime = FOREGROUND_CHECK_INTERVAL;
+        }
+      } else {
+        // App in background
+        intervalTime = BACKGROUND_CHECK_INTERVAL;
+      }
+
+      // Set up the interval
+      checkIntervalRef.current = setInterval(() => {
+        // Using a function that ignores the Promise
+        checkAuth().catch((err) =>
+          logger.error("setupCheckInterval", "Error checking auth", err),
+        );
+      }, intervalTime);
+    },
+    [isActive, checkAuth],
+  );
+
+  // Handle app state changes (foreground/background)
+  useEffect(() => {
+    const handleAppStateChange = (nextAppState: AppStateStatus) => {
+      if (
+        appState.current.match(/inactive|background/) &&
+        nextAppState === "active"
+      ) {
+        // App has come to the foreground - check auth with a delay to allow app to fully resume
+        setTimeout(() => {
+          checkAuth().catch((err) =>
+            logger.error("handleAppStateChange", "Error checking auth", err),
+          );
+        }, 300);
+      }
+
+      appState.current = nextAppState;
+      setupCheckInterval(nextAppState);
+    };
+
+    const subscription = AppState.addEventListener(
+      "change",
+      handleAppStateChange,
+    );
+
+    // Initial setup with a slight delay to ensure app is fully initialized
+    setTimeout(() => {
+      setupCheckInterval(appState.current);
+    }, 300);
+
+    return () => {
+      subscription.remove();
+      if (checkIntervalRef.current) {
+        clearInterval(checkIntervalRef.current);
+      }
+    };
+  }, [setupCheckInterval, checkAuth]);
+
+  // Track user interaction
+  useEffect(() => {
+    const interactionStartListener =
```

Comment:
Is this `interactionStartListener` being effectively used? It appears this is being created here and removed on the useEffect clean-up function but I don't see any other reference to it